### ISSUE TITLE
Server side defaults not diffing correctly

### ISSUE THREAD

**Issue by Dhouti** [zalando/postgres-operator#555]
This issue presented itself when switching kubernetes versions.

Kubernetes Server Version: v1.13.5

```
level=debug msg="spec diff between old and new statefulsets:
Template.Spec.Containers[0].TerminationMessagePath: "/dev/termination-log" != ""
Template.Spec.Containers[0].TerminationMessagePolicy: "File" != ""
Template.Spec.RestartPolicy: "Always" != ""
Template.Spec.DNSPolicy: "ClusterFirst" != ""
Template.Spec.DeprecatedServiceAccount: "operator" != ""
Template.Spec.SecurityContext: &v1.PodSecurityContext{} != nil
Template.Spec.SchedulerName: "default-scheduler" != ""
Template.Spec.Tolerations: []v1.Toleration(nil) != []v1.Toleration{}
VolumeClaimTemplates[0].Spec.VolumeMode: &"Filesystem" != nil
VolumeClaimTemplates[0].Status.Phase: "Pending" != ""
PodManagementPolicy: "OrderedReady" != ""
UpdateStrategy.Type: "OnDelete" != ""
RevisionHistoryLimit: &int32(10) != nil"
```

The "old" values in question are defaulted by the kubernetes server.
This is causing the statefulset to be recreated every time it is synced.

**Comment by knan-nrk** [comment]
Also hits us on every sync. Kubernetes version 1.13.3.

`time="2019-05-10T08:14:53Z" level=info msg="statefulset \"mediakvern-experimental/mediakvernexperimental-db\" is not in the desired state and needs to be updated" cluster-name=mediakvern-experimental/mediakvern-experimental-db pkg=cluster worker=0

time="2019-05-10T08:14:53Z" level=debug msg="spec diff between old and new statefulsets: \nTemplate.Spec.Containers[0].TerminationMessagePath: \"/dev/termination-log\" != \"\"\nTemplate.Spec.Containers[0].TerminationMessagePolicy: \"File\" != \"\"\nTemplate.Spec.RestartPolicy: \"Always\" != \"\"\nTemplate.Spec.DNSPolicy: \"ClusterFirst\" != \"\"\nTemplate.Spec.DeprecatedServiceAccount: \"zalando-postgres-operator\" != \"\"\nTemplate.Spec.SchedulerName: \"default-scheduler\" != \"\"\nTemplate.Spec.Tolerations: []v1.Toleration(nil) != []v1.Toleration{}\nVolumeClaimTemplates[0].Spec.VolumeMode: &\"Filesy
stem\" != nil\nVolumeClaimTemplates[0].Status.Phase: \"Pending\" != \"\"\nRevisionHistoryLimit: &int32(10) != nil\n" cluster-name=mediakvern-experimental/mediakvern-experimental-db pkg=cluster worker=0`

**Comment by Jan-M** [comment]
Thanks for reporting this issue and reraising it. We are seeing this in 1.13 test environments too and are looking into the resolution.

**Comment by sdudoladov** [comment]
closed by #563
